### ISSUE TITLE
"Couldn't find a debug adapter descriptor for debug type 'android-webview' extension might have failed to activate)"

### ISSUE THREAD

**Issue by vmonteco** [mpotthoff/vscode-android-webview-debug#9]
Hello,

**TL;DR:**
I'm trying to install the android webview debugger but trying to use it always  issue an error saying that the JS debugger wasn't loaded.

---

I've tried to install the extension but it fails to launch.
I installed the JS debugger and the android-webview-debugging tool.

I've added the following configuration in my `launch.json` :

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "android-webview",
            "request": "attach",
            "name": "Attach to Android WebView",
            "device": "XXXXXXXXXXXXXX",
            "webRoot": "${workspaceFolder}",
            "urlFilter": "*"
        }
    ]
}
```

My device is up and running and I can see it through the `adb -H 10.0.2.2 -P 5037 devices` command (I access my working environment through a remote ssh session in a virtualbox VM).

But when I try to start a debugging session, I get a popup with the following :

> Couldn't find a debug adapter descriptor for debug type 'android-webview' (extension might have failed to activate).

and the console of dev tools gives me the following :

```
Cannot activate the 'Android WebView Debugging' extension because it depends on the 'JavaScript Debugger' extension, which is not loaded. Would you like to reload the window to load the extension?
```

Reloading VSC doesn't seem to change anything.

I hope I post this in the right place, I searched in the previous issues and I found nothing that seemed related.

How could I investigate/fix that ?

**Comment by mpotthoff** [comment]
Hey!

So that is actually a use case of which I haven't thought about before.
The reason for the error you are seeing is that the extension runs on the UI side instead of the workspace side (which I only configured because of some problems with the old chrome debugger extension). 

In the latest version 1.2.2 (just published) I changed it so that the extension always runs on the workspace side (which now seems to be the preferred way with the new js-debug extension). That should fix the error.

The other thing you seem to need is a way to specify additional arguments for adb (`-H 10.0.2.2 -P 5037` in your case). I added a new configuration parameter `android-webview-debug.adbArgs` which you can use to set them. (`"android-webview-debug.adbArgs": ["-H", "10.0.2.2", "-P", "5037"]`)

This should solve all your problems. 😄

**Comment by vmonteco** [comment]
Hello !
Thank you for this quick and detailed answer which is very appreciated !
Unfortunately I don't have the device I was working with with me right now, but next time I'm at work I check that. :)
I'll keep you posted to confirm it works !

**Comment by vmonteco** [comment]
Hello @mpotthoff !
I finally tested with my smartphone.
I added the following configuration as suggested :
```
        {
            "type": "android-webview",
            "request": "attach",
            "name": "FP3",
            "device": "`XXXXXXXXXXXXX",
            "webRoot": "${workspaceFolder}",
            "urlFilter": "https://xxx.yyy/*"
        }
```

I also added `["-H", "10.0.2.2", "-P", "5037"]` as value for the suggested settings and I guess it worked since I didn't have the "adb: unknown command ["-H", "10.0.2.2", "-P", "5037"]" notification anymore (I entered it the wrong way previously).

Yet, when trying to run the debugger, I unexpectedly get the following popup :
![image](https://user-images.githubusercontent.com/11329528/156387044-738c3c9f-cae7-4c2c-b8a6-19d5b1722af5.png)

Have you ever met such problem ?

**Comment by vmonteco** [comment]
I'd also add that VSC succeeded at detecting my webview since it prompted me to select one and "Chrome" and other apps were amongst the choices.

**Comment by vmonteco** [comment]
ERRATUM :
It looks like [it's solved by the nightly release.](https://github.com/microsoft/vscode-js-debug/issues/590#issuecomment-657561476)

But now I have the `This adb server's $ADB_VENDOR_KEYS is not set` error.

**Comment by mpotthoff** [comment]
Hey @vmonteco !

Regarding the first issue with the screenshot:
I don't think this is related to the nightly release as the mentioned issue is already 2 years old 😋  Did you have `debug.javascript.usePreview` explicitly set to false by any chance (The default is true these days)? In that case I still try to use the old chrome debugger extension which runs on the UI side (which would break the connection setup on remote debugging since I have moved everything to the workspace side).

Regarding the second issue:
This sounds like a problem with the adb server. Does `adb -H 10.0.2.2 -P 5037 devices` still work when you run it manually in the terminal? It seems like revoking the debugging authorization sometimes does the trick: https://stackoverflow.com/questions/31638582/android-adb-devices-unauthorized
